Source/ARTLocalDevice.m

```diff
@@ -75,21 +75,20 @@ + (ARTLocalDevice *)load:(NSString *)clientId storage:(id<ARTDeviceStorage>)stor
     NSString *deviceId = [storage objectForKey:ARTDeviceIdKey];
     NSString *deviceSecret = deviceId == nil ? nil : [storage secretForDevice:deviceId];
     
-    if (deviceId == nil || deviceSecret == nil) { // generate both at the same time
-        deviceId = [self generateId];
-        deviceSecret = [self generateSecret];
-        
-        [storage setObject:deviceId forKey:ARTDeviceIdKey];
-        [storage setSecret:deviceSecret forDevice:deviceId];
-    }
-    
-    device.id = deviceId;
+    device.id = deviceId ?: [self.class generateId]; // temporarily ignore RSH8a/b, see https://github.com/ably/ably-cocoa/pull/1847#discussion_r1441954284 thread
```

Comment:
I think that it would be better to create an issue for this in https://github.com/ably/specification (with a proper description) and link to that.

Reply:
https://github.com/ably/specification/issues/180

Reply:
https://github.com/ably/ably-cocoa/pull/1847/commits/79e741b269e7ab7486292fc0acc09b5146c2badf